drf_extra_fields/fields.py

```diff
@@ -221,13 +221,23 @@ def to_representation(self, value):
         """
         Range instances -> dicts of primitive datatypes.
         """
-        if value.isempty:
-            return {'empty': True}
-        lower = self.child.to_representation(value.lower) if value.lower is not None else None
-        upper = self.child.to_representation(value.upper) if value.upper is not None else None
+        if isinstance(value, dict):
```

Comment:
If you provide a dict object to a serializer as `instance` the value is not converted to internal value.  (FooSerializer(data))

`data` of the serializer is run through validation.  In this case `value` is a range object. (FooSerializer(data=data))

Reply:
We may refactor this method as follows:

```python
def to_representation(self, value):
    """
    Range instances -> dicts of primitive datatypes.
    """
    if isinstance(value, dict) and not value:
        return value
    elif value.isempty:
        return {"empty": True}

    lower = value.get("lower") if isinstance(value, dict) else value.lower
    upper = value.get("upper") if isinstance(value, dict) else value.upper
    bounds = value.get("bounds") if isinstance(value, dict) else value._bounds

    return {"lower":  self.child.to_representation(lower) if lower is not None else None,
            "upper": self.child.to_representation(upper) if upper is not None else None,
            "bounds": bounds}
```

or 

```python
def to_representation(self, value):
    """
    Range instances -> dicts of primitive datatypes.
    """
    if isinstance(value, dict):
        if not value:
            return value

        lower = value.get("lower")
        upper = value.get("upper")
        bounds = value.get("bounds")
    else:
        if value.isempty:
            return {"empty": True}
        
        lower = value.lower
        upper = value.upper
        bounds = value._bounds

    return {"lower":  self.child.to_representation(lower) if lower is not None else None,
            "upper": self.child.to_representation(upper) if upper is not None else None,
            "bounds": bounds}
```

Reply:
Thanks, updated.